plugin/src/main/java/org/opensearch/ml/action/update_cache/UpdateModelCacheTransportAction.java

```diff
@@ -106,20 +106,15 @@ private MLUpdateModelCacheNodeResponse createUpdateModelCacheNodeResponse(
         MLUpdateModelCacheNodesRequest mlUpdateModelCacheNodesRequest
     ) {
         String modelId = mlUpdateModelCacheNodesRequest.getModelId();
```

Comment:
Why not use the mlModelManger to updateModelCache directly? What is the reason behind that we have to wrap this into a transport action call? I see that it's not taking any inputs from Users or Rest Layer.

Reply:
If this is a single node cluster then yes. However this is not the case in multi-node cluster. This is because we have to use the `NodeRequest` to ensure that we send the request to all necessary nodes.